infra/Pos/DHT/Real/Real.hs

```diff
@@ -210,11 +215,11 @@ kademliaJoinNetwork
     -> [NetworkAddress]
     -> m ()
 kademliaJoinNetwork _ [] = throw AllPeersUnavailable
-kademliaJoinNetwork inst nodes =
-    waitAnyUnexceptional (map (kademliaJoinNetwork' inst) nodes) >>= handleRes
-  where
-    handleRes (Just _) = pure ()
-    handleRes _        = throw AllPeersUnavailable
+kademliaJoinNetwork inst (node : nodes) = do
+    outcome <- try (kademliaJoinNetwork' inst node)
+    case outcome of
+        Left (e :: DHTException) -> kademliaJoinNetwork inst nodes
+        Right _ -> return ()
```

Comment:
I think it's better this way: no sense trying to join the network at multiple points concurrently. In fact, I suspect it could lead to a false positive identifier clash.